timetracker/vms/models.py

```diff
@@ -732,6 +748,39 @@ def is_approved(self):
         """
         return hasattr(self, 'approval')
 
+    @property
+    def total_time(self):
+        """
+        Get the total time from this time record.
+
+        Returns:
+            The total time in this time record
+            that the employee has worked in seconds,
+            rounded to the nearest 15 minute increment.
+        """
+        hours = datetime.timedelta(0)
+        if self.time_end:
+            hours += self.time_end - self.time_start
+            seconds_worked = hours.total_seconds()
+            total_hours = seconds_worked / (60 * 60)
+            return math.floor(total_hours * 2) / 2
+        else:
+            return None
+
+    @property
+    def projected_earnings(self):
+        """
+        Get the projected earning from this time record.
+
+        Returns:
+            The total time multiplied by
+            the pay rate for this job.
+        """
+        if self.total_time and self.job and self.job.pay_rate:
+            return float(self.total_time) * float(self.job.pay_rate)
```

Comment:
Time records have a copy of the pay rate in case the pay rate for the job changes (we don't want to update the pay rate of previous time records).

```suggestion
            return self.total_time * self.pay_rate
```